pkg/controller/validationscache.go

```diff
@@ -64,35 +72,45 @@ func (vc *validationCache) store(obj client.Object, outcome validations.Validati
 	)
 }
 
+// drain overwrites the current instance with an empty one
+// all keys and values will be lost
 func (vc *validationCache) drain() {
 	*vc = validationCache{}
 }
 
+// remove deletes a key, and its value, from the instance
+// it uses given object to search for the validationKey
 func (vc *validationCache) remove(obj client.Object) {
 	key := newValidationKey(obj)
 	vc.removeKey(key)
 }
 
+// removeKey deletes a key, and its value, from the instance
 func (vc *validationCache) removeKey(key validationKey) {
 	delete(*vc, key)
 }
 
+// retrieve returns the value, if exists, within a key
+// it uses given object to search for the validationKey
+// it returns a second parameter to check if the key existed in the instance
```

Comment:
```suggestion
// retrieve returns a tuple of 'validationResource' (if present)
// and 'ok' which returns 'true' if a 'validationResource' exists
// for the given 'Object' and 'false' otherwise.
```